### PR TITLE
Mostly switch to chrono

### PR DESCRIPTION
**Pull request by Kixiron**
Most of the stuff can be switched to `chrono`, but iron-related timeouts require `time` and won't be able to be removed until `iron` is.

### CODE REVIEW

**Comment by Kixiron** [comment]
This is currently kinda broken, but it's caused me so much pain that I'm hoping someone else knows what the hell they're doing

**Comment by jyn514** [comment]
This is the relevant error message: https://github.com/rust-lang/docs.rs/runs/718596017#step:11:1876

<details><summary>backtrace</summary>

```
2020-05-28T23:12:12.2894420Z [2020-05-28T23:12:12Z DEBUG cratesfyi::db::add_package] Adding package into database
2020-05-28T23:12:12.3107644Z [2020-05-28T23:12:12Z DEBUG cratesfyi::db::add_package] Adding build into database
2020-05-28T23:12:13.1011236Z thread '<unnamed>' panicked at 'a Display implementation returned an error unexpectedly: Error', /rustc/8d69840ab92ea7f4d323420088dd8c9775f180cd/src/liballoc/string.rs:2165:9
2020-05-28T23:12:13.1011570Z stack backtrace:
2020-05-28T23:12:13.1011763Z    0: backtrace::backtrace::libunwind::trace
2020-05-28T23:12:13.1012289Z              at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.44/src/backtrace/libunwind.rs:86
2020-05-28T23:12:13.1012461Z    1: backtrace::backtrace::trace_unsynchronized
2020-05-28T23:12:13.1012851Z              at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.44/src/backtrace/mod.rs:66
2020-05-28T23:12:13.1013013Z    2: std::sys_common::backtrace::_print_fmt
2020-05-28T23:12:13.1013152Z              at src/libstd/sys_common/backtrace.rs:78
2020-05-28T23:12:13.1013308Z    3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
2020-05-28T23:12:13.1013463Z              at src/libstd/sys_common/backtrace.rs:59
2020-05-28T23:12:13.1013605Z    4: core::fmt::write
2020-05-28T23:12:13.1013737Z              at src/libcore/fmt/mod.rs:1063
2020-05-28T23:12:13.1013857Z    5: std::io::Write::write_fmt
2020-05-28T23:12:13.1013991Z              at src/libstd/io/mod.rs:1426
2020-05-28T23:12:13.1014428Z    6: std::sys_common::backtrace::_print
2020-05-28T23:12:13.1014565Z              at src/libstd/sys_common/backtrace.rs:62
2020-05-28T23:12:13.1014705Z    7: std::sys_common::backtrace::print
2020-05-28T23:12:13.1014845Z              at src/libstd/sys_common/backtrace.rs:49
2020-05-28T23:12:13.1014985Z    8: std::panicking::default_hook::{{closure}}
2020-05-28T23:12:13.1015109Z              at src/libstd/panicking.rs:204
2020-05-28T23:12:13.1015245Z    9: std::panicking::default_hook
2020-05-28T23:12:13.1015380Z              at src/libstd/panicking.rs:224
2020-05-28T23:12:13.1015515Z   10: std::panicking::rust_panic_with_hook
2020-05-28T23:12:13.1015649Z              at src/libstd/panicking.rs:470
2020-05-28T23:12:13.1015781Z   11: rust_begin_unwind
2020-05-28T23:12:13.1015911Z              at src/libstd/panicking.rs:378
2020-05-28T23:12:13.1016037Z   12: core::panicking::panic_fmt
2020-05-28T23:12:13.1016170Z              at src/libcore/panicking.rs:85
2020-05-28T23:12:13.1016312Z   13: core::option::expect_none_failed
2020-05-28T23:12:13.1016446Z              at src/libcore/option.rs:1211
2020-05-28T23:12:13.1016582Z   14: core::result::Result<T,E>::expect
2020-05-28T23:12:13.1016822Z              at /rustc/8d69840ab92ea7f4d323420088dd8c9775f180cd/src/libcore/result.rs:961
2020-05-28T23:12:13.1016985Z   15: <T as alloc::string::ToString>::to_string
2020-05-28T23:12:13.1017138Z              at /rustc/8d69840ab92ea7f4d323420088dd8c9775f180cd/src/liballoc/string.rs:2165
2020-05-28T23:12:13.1017287Z   16: <cratesfyi::web::releases::Release as serde::ser::Serialize>::serialize
2020-05-28T23:12:13.1017437Z              at src/web/releases.rs:61
2020-05-28T23:12:13.1017582Z   17: serde::ser::impls::<impl serde::ser::Serialize for &T>::serialize
2020-05-28T23:12:13.1018038Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.110/src/ser/impls.rs:390
2020-05-28T23:12:13.1018211Z   18: serde::ser::impls::<impl serde::ser::Serialize for &T>::serialize
2020-05-28T23:12:13.1018638Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.110/src/ser/impls.rs:390
2020-05-28T23:12:13.1018811Z   19: serde::ser::impls::<impl serde::ser::Serialize for &T>::serialize
2020-05-28T23:12:13.1019227Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.110/src/ser/impls.rs:390
2020-05-28T23:12:13.1019384Z   20: serde_json::value::to_value
2020-05-28T23:12:13.1019779Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_json-1.0.53/src/value/mod.rs:964
2020-05-28T23:12:13.1019961Z   21: <serde_json::value::ser::SerializeVec as serde::ser::SerializeSeq>::serialize_element
2020-05-28T23:12:13.1020384Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_json-1.0.53/src/value/ser.rs:313
2020-05-28T23:12:13.1020548Z   22: serde::ser::Serializer::collect_seq
2020-05-28T23:12:13.1020951Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.110/src/ser/mod.rs:1283
2020-05-28T23:12:13.1021134Z   23: serde::ser::impls::<impl serde::ser::Serialize for alloc::vec::Vec<T>>::serialize
2020-05-28T23:12:13.1021556Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.110/src/ser/impls.rs:198
2020-05-28T23:12:13.1021725Z   24: serde::ser::impls::<impl serde::ser::Serialize for &T>::serialize
2020-05-28T23:12:13.1022133Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.110/src/ser/impls.rs:390
2020-05-28T23:12:13.1022297Z   25: serde::ser::impls::<impl serde::ser::Serialize for &T>::serialize
2020-05-28T23:12:13.1022689Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.110/src/ser/impls.rs:390
2020-05-28T23:12:13.1022849Z   26: serde_json::value::to_value
2020-05-28T23:12:13.1023252Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_json-1.0.53/src/value/mod.rs:964
2020-05-28T23:12:13.1023508Z   27: <serde_json::value::ser::SerializeMap as serde::ser::SerializeMap>::serialize_value
2020-05-28T23:12:13.1023948Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_json-1.0.53/src/value/ser.rs:410
2020-05-28T23:12:13.1024115Z   28: serde::ser::SerializeMap::serialize_entry
2020-05-28T23:12:13.1024522Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.110/src/ser/mod.rs:1823
2020-05-28T23:12:13.1024704Z   29: <serde_json::value::ser::SerializeMap as serde::ser::SerializeStruct>::serialize_field
2020-05-28T23:12:13.1025132Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_json-1.0.53/src/value/ser.rs:622
2020-05-28T23:12:13.1025298Z   30: <cratesfyi::web::page::Page<T> as serde::ser::Serialize>::serialize
2020-05-28T23:12:13.1025429Z              at src/web/page.rs:135
2020-05-28T23:12:13.1025577Z   31: serde::ser::impls::<impl serde::ser::Serialize for &T>::serialize
2020-05-28T23:12:13.1025986Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.110/src/ser/impls.rs:390
2020-05-28T23:12:13.1026147Z   32: serde_json::value::to_value
2020-05-28T23:12:13.1026616Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_json-1.0.53/src/value/mod.rs:964
2020-05-28T23:12:13.1026784Z   33: handlebars::context::to_json
2020-05-28T23:12:13.1027204Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/handlebars-0.29.1/src/context.rs:209
2020-05-28T23:12:13.1027773Z   34: handlebars_iron::middleware::Template::new
2020-05-28T23:12:13.1028566Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/handlebars-iron-0.25.2/src/middleware.rs:29
2020-05-28T23:12:13.1028922Z   35: cratesfyi::web::page::Page<T>::to_resp
2020-05-28T23:12:13.1029204Z              at src/web/page.rs:107
2020-05-28T23:12:13.1029499Z   36: cratesfyi::web::releases::releases_handler
2020-05-28T23:12:13.1029790Z              at src/web/releases.rs:376
2020-05-28T23:12:13.1030096Z   37: cratesfyi::web::releases::recent_releases_handler
2020-05-28T23:12:13.1030391Z              at src/web/releases.rs:398
2020-05-28T23:12:13.1030685Z   38: core::ops::function::Fn::call
2020-05-28T23:12:13.1030989Z              at /rustc/8d69840ab92ea7f4d323420088dd8c9775f180cd/src/libcore/ops/function.rs:72
2020-05-28T23:12:13.1031298Z   39: <F as iron::middleware::Handler>::handle
2020-05-28T23:12:13.1031845Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/iron-0.5.1/src/middleware/mod.rs:312
2020-05-28T23:12:13.1032187Z   40: <alloc::boxed::Box<dyn iron::middleware::Handler> as iron::middleware::Handler>::handle
2020-05-28T23:12:13.1032737Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/iron-0.5.1/src/middleware/mod.rs:318
2020-05-28T23:12:13.1033742Z   41: <cratesfyi::web::metrics::RequestRecorder as iron::middleware::Handler>::handle
2020-05-28T23:12:13.1034015Z              at src/web/metrics.rs:182
2020-05-28T23:12:13.1034276Z   42: <alloc::boxed::Box<dyn iron::middleware::Handler> as iron::middleware::Handler>::handle
2020-05-28T23:12:13.1035341Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/iron-0.5.1/src/middleware/mod.rs:318
2020-05-28T23:12:13.1035639Z   43: <alloc::boxed::Box<dyn iron::middleware::Handler> as iron::middleware::Handler>::handle
2020-05-28T23:12:13.1036126Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/iron-0.5.1/src/middleware/mod.rs:318
2020-05-28T23:12:13.1036393Z   44: router::router::Router::handle_method
2020-05-28T23:12:13.1036854Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/router-0.5.1/src/router.rs:199
2020-05-28T23:12:13.1037120Z   45: <router::router::Router as iron::middleware::Handler>::handle
2020-05-28T23:12:13.1037588Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/router-0.5.1/src/router.rs:212
2020-05-28T23:12:13.1037858Z   46: <alloc::boxed::Box<dyn iron::middleware::Handler> as iron::middleware::Handler>::handle
2020-05-28T23:12:13.1038476Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/iron-0.5.1/src/middleware/mod.rs:318
2020-05-28T23:12:13.1038744Z   47: iron::middleware::Chain::continue_from_handler
2020-05-28T23:12:13.1039215Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/iron-0.5.1/src/middleware/mod.rs:282
2020-05-28T23:12:13.1039478Z   48: iron::middleware::Chain::continue_from_before
2020-05-28T23:12:13.1039945Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/iron-0.5.1/src/middleware/mod.rs:276
2020-05-28T23:12:13.1043149Z   49: <iron::middleware::Chain as iron::middleware::Handler>::handle
2020-05-28T23:12:13.1043861Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/iron-0.5.1/src/middleware/mod.rs:200
2020-05-28T23:12:13.1044145Z   50: <alloc::boxed::Box<dyn iron::middleware::Handler> as iron::middleware::Handler>::handle
2020-05-28T23:12:13.1044633Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/iron-0.5.1/src/middleware/mod.rs:318
2020-05-28T23:12:13.1045045Z   51: <cratesfyi::web::CratesfyiHandler as iron::middleware::Handler>::handle::{{closure}}
2020-05-28T23:12:13.1045298Z              at src/web/mod.rs:145
2020-05-28T23:12:13.1045530Z   52: core::result::Result<T,E>::or_else
2020-05-28T23:12:13.1045769Z              at /rustc/8d69840ab92ea7f4d323420088dd8c9775f180cd/src/libcore/result.rs:797
2020-05-28T23:12:13.1046623Z   53: <cratesfyi::web::CratesfyiHandler as iron::middleware::Handler>::handle
2020-05-28T23:12:13.1046872Z              at src/web/mod.rs:143
2020-05-28T23:12:13.1047322Z   54: <iron::iron::RawHandler<H> as hyper::server::Handler>::handle
2020-05-28T23:12:13.1047848Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/iron-0.5.1/src/iron.rs:176
2020-05-28T23:12:13.1048108Z   55: hyper::server::Worker<H>::keep_alive_loop
2020-05-28T23:12:13.1048588Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.10.16/src/server/mod.rs:340
2020-05-28T23:12:13.1048849Z   56: hyper::server::Worker<H>::handle_connection
2020-05-28T23:12:13.1049329Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.10.16/src/server/mod.rs:282
2020-05-28T23:12:13.1049590Z   57: hyper::server::handle::{{closure}}
2020-05-28T23:12:13.1050049Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.10.16/src/server/mod.rs:242
2020-05-28T23:12:13.1050314Z   58: hyper::server::listener::spawn_with::{{closure}}
2020-05-28T23:12:13.1050789Z              at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.10.16/src/server/listener.rs:50
2020-05-28T23:12:13.1051057Z note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

</details>

**Comment by jyn514** [comment]
Looks like the `impl Display for Release` returned an error.

**Comment by Kixiron** [comment]
My favorite error, `Error`! I’ll take a stab tomorrow

**Comment by Nemo157** [comment]
Using `Local` seems wrong, nothing should care about what the timezone of the server is (except _maybe_ the "cronjobs", though I think making them UTC based as well would make sense).

**Comment by Kixiron** [comment]
While I agree that using UTC would be preferable, it requires changing every row of the DB that uses `TIMESTAMP` to use `TIMESTAMPZ` which I don't really know how to do

**Comment by Nemo157** [comment]
The database is already using `timestamp without time zone` correctly, I assume there's just weirdness happening currently if you are running in non-UTC, but I couldn't see any way to view timestamps in the UI to confirm.

**Comment by Kixiron** [comment]
The only place I know of to view timestamps is on [builds](https://docs.rs/crate/regex/1.3.9/builds/255289), and you can't serialize a `DateTime<_>` from our current DB, only a `NiaveDateTime` since `DateTime`s have timezones and `NaiveDateTime` doesn't

File: src/index/api.rs [hunk 1]
```diff
@@ -70,11 +68,8 @@ fn get_release_time_yanked_downloads(pkg: &MetadataPackage) -> Result<(time::Tim
                 .get("created_at")
                 .and_then(|c| c.as_str())
                 .ok_or_else(|| err_msg("Not a JSON object"))?;
-            release_time = Some(
-                time::strptime(release_time_raw, "%Y-%m-%dT%H:%M:%S")
-                    .unwrap()
-                    .to_timespec(),
-            );
+            release_time =
+                Some(NaiveDateTime::parse_from_str(release_time_raw, "%Y-%m-%dT%H:%M:%S").unwrap());
```
[the remark this answers is not in the document]
Comment (Nemo157) [reply src/index/api.rs:72]: In fact, this should be something like

```suggestion
                Some(NaiveDateTime::parse_from_str(release_time_raw, "%Y-%m-%dT%H:%M:%S%.f%:z")?);
```

in case the crates.io API returns an invalid value

File: src/index/api.rs [hunk 2]
```diff
@@ -70,10 +68,11 @@ fn get_release_time_yanked_downloads(pkg: &MetadataPackage) -> Result<(time::Tim
                 .get("created_at")
                 .and_then(|c| c.as_str())
                 .ok_or_else(|| err_msg("Not a JSON object"))?;
+
             release_time = Some(
-                time::strptime(release_time_raw, "%Y-%m-%dT%H:%M:%S")
+                DateTime::parse_from_str(release_time_raw, "%Y-%m-%dT%H:%M:%S%.f%:z")
                     .unwrap()
```
Comment (Nemo157) [inline src/index/api.rs:74]: It'd be nice to remove this unwrap so that we don't crash if the crates.io API returns something bad.

File: src/web/file.rs [hunk 3]
```diff
@@ -27,9 +27,10 @@ impl File {
             .headers
             .set(ContentType(self.0.mime.parse().unwrap()));
         response.headers.set(CacheControl(cache));
-        response
-            .headers
-            .set(LastModified(HttpDate(time::at(self.0.date_updated))));
+        // FIXME: This is so horrible
+        response.headers.set(LastModified(HttpDate(
+            time::strptime(&self.0.date_updated.format("%+").to_string(), "%Y-%m-%dT%T").unwrap(),
+        )));
```
Comment (jyn514) [inline src/web/file.rs:33]: Can you add a round trip test for this?

File: src/db/add_package.rs [hunk 4]
```diff
@@ -86,7 +86,7 @@ pub(crate) fn add_package_into_database(
         &[
             &crate_id,
             &metadata_pkg.version,
-            &cratesio_data.release_time,
+            &cratesio_data.release_time.naive_utc(),
```
Comment (jyn514) [inline src/db/add_package.rs:89]: What does `naive` mean here?

File: src/index/api.rs [hunk 5]
```diff
@@ -65,15 +63,15 @@ fn get_release_time_yanked_downloads(pkg: &MetadataPackage) -> Result<(time::Tim
...
                 .get("created_at")
                 .and_then(|c| c.as_str())
                 .ok_or_else(|| err_msg("Not a JSON object"))?;
+
             release_time = Some(
-                time::strptime(release_time_raw, "%Y-%m-%dT%H:%M:%S")
-                    .unwrap()
-                    .to_timespec(),
+                DateTime::parse_from_str(release_time_raw, "%Y-%m-%dT%H:%M:%S%.f%:z")?
+                    .with_timezone(&Utc),
```
Comment (jyn514) [inline src/index/api.rs:74]: Were past builds uploaded using UTC timezones? I'm not sure this is correct for existing builds.

File: src/storage/s3.rs [hunk 6]
```diff
@@ -100,9 +100,13 @@ impl<'a> S3Backend<'a> {
...
-fn parse_timespec(raw: &str) -> Result<Timespec, Error> {
-    const TIME_FMT: &str = "%a, %d %b %Y %H:%M:%S %Z";
-    Ok(time::strptime(raw, TIME_FMT)?.to_timespec())
+fn parse_timespec(mut raw: &str) -> Result<DateTime<Utc>, Error> {
+    raw = raw.trim_end_matches(" GMT");
+
+    Ok(DateTime::from_utc(
+        NaiveDateTime::parse_from_str(raw, "%a, %d %b %Y %H:%M:%S")?,
+        Utc,
+    ))
```
Comment (jyn514) [inline src/storage/s3.rs:109]: Can you add a test for this?

File: src/utils/github_updater.rs [hunk 7]
```diff
@@ -121,12 +122,11 @@ fn get_github_fields(path: &str) -> Result<GitHubFields> {
...
         issues: obj.get("open_issues").and_then(|d| d.as_i64()).unwrap_or(0),
-        last_commit: time::strptime(
+        last_commit: DateTime::parse_from_rfc3339(
             obj.get("pushed_at").and_then(|d| d.as_str()).unwrap_or(""),
-            "%Y-%m-%dT%H:%M:%S",
         )
-        .unwrap_or_else(|_| time::now())
-        .to_timespec(),
+        .map(|datetime| datetime.with_timezone(&Utc))
+        .unwrap_or_else(|_| Utc::now()),
```
Comment (jyn514) [inline src/utils/github_updater.rs:129]: Can you summarize why we're adding a timezone here? It looks like this changed from the previous behavior.

File: src/utils/release_activity_updater.rs [hunk 8]
```diff
@@ -37,11 +37,10 @@ pub fn update_release_activity() -> Result<()> {
 
         let release_count: i64 = rows.get(0).get(0);
         let failure_count: i64 = failures_count_rows.get(0).get(0);
-        let now = now();
-        let date = now - Duration::days(day);
+        let now = Utc::now().naive_utc();
+        let date = now.checked_sub_signed(Duration::days(day)).unwrap();
```
Comment (jyn514) [inline src/utils/release_activity_updater.rs:41]: Why bother using `checked` if you're just going to `unwrap`?

File: src/web/builds.rs [hunk 9]
```diff
@@ -27,10 +28,7 @@ impl Serialize for Build {
         state.serialize_field("rustc_version", &self.rustc_version)?;
         state.serialize_field("cratesfyi_version", &self.cratesfyi_version)?;
         state.serialize_field("build_status", &self.build_status)?;
-        state.serialize_field(
-            "build_time",
-            &time::at(self.build_time).rfc3339().to_string(),
-        )?;
+        state.serialize_field("build_time", &self.build_time.format("%+").to_string())?;
```
Comment (jyn514) [inline src/web/builds.rs:31]: What does `%+` mean? Can you add that as a comment?

File: src/utils/release_activity_updater.rs (hunk 8 above)
Comment (Kixiron) [reply src/utils/release_activity_updater.rs:41]: It's the only option for chrono types

File: src/web/builds.rs (hunk 9 above)
Comment (Kixiron) [reply src/web/builds.rs:31]: `%+` just means a RFC 3339 timestamp, it's in the [chrono docs](https://docs.rs/chrono/0.4.11/chrono/format/strftime/index.html)

File: src/utils/github_updater.rs (hunk 7 above)
Comment (Kixiron) [reply src/utils/github_updater.rs:129]: `time::now()` gets the current timestamp which is then turned into a zoneless time, we're doing the same thing but keeping the zone

File: src/db/add_package.rs (hunk 4 above)
Comment (Kixiron) [reply src/db/add_package.rs:89]: `Naive`/`naive` stuff in `chrono` are times without time zones, and we store a timestamp without a timezone in the database

**Comment by Kixiron** [comment]
The reason we're moving to chrono is that it's newer, more widely used and the successor to `time`. Chrono gives a lot of flexibility and has good docs, it's an overall bonus

File: src/index/api.rs (hunk 5 above)
Comment (Nemo157) [reply src/index/api.rs:74]: That might depend on what the server is set to (can you check that?). Hopefully it's UTC and everything just works out, otherwise a migration to update the times in the database could be necessary.

File: src/storage/s3.rs (hunk 6 above)
Comment (Kixiron) [reply src/storage/s3.rs:109]: There already are some [here](https://github.com/rust-lang/docs.rs/pull/781/files/8ef89945beaf1de95a589228e7e980188526fa1d#diff-c3a2a09475e9683a0e82b9a1d6f5e0b9R148-R156), do you want more?

File: src/web/file.rs (hunk 3 above)
Comment (Kixiron) [reply src/web/file.rs:33]: That conversion is by nature lossy because `time` has a lower resolution than `chrono`, but it also probably doesn't matter to terribly much because it doesn't do a whole lot

File: src/index/api.rs (hunk 5 above)
Comment (jyn514) [reply src/index/api.rs:74]: The server is in UTC as far as I can tell.
```
$ date
Wed Jun  3 01:06:35 UTC 2020
```

File: src/storage/s3.rs (hunk 6 above)
Comment (jyn514) [reply src/storage/s3.rs:109]: Those look great, thanks.

File: src/web/file.rs (hunk 3 above)
Comment (jyn514) [reply src/web/file.rs:33]: I disagree, unless we test it we don't know that the string formats match. It should be a simple test anyway.

> time has a lower resolution than chrono

That shouldn't affect it? The time starts as `1234.456` or whatever, gets a few zeros added at the end, then the zeros are removed again.

File: src/utils/release_activity_updater.rs (hunk 8 above)
Comment (jyn514) [reply src/utils/release_activity_updater.rs:41]: `DateTime` implements `Sub`: https://docs.rs/chrono/0.4.11/chrono/struct.DateTime.html#impl-Sub%3CDuration%3E

Comment (Kixiron) [reply src/utils/release_activity_updater.rs]: Whoops, didn't see that

File: src/web/builds.rs (hunk 9 above)
Comment (jyn514) [reply src/web/builds.rs:31]: Let's add a comment anyway in case the docs or behavior ever change.
